### PR TITLE
Add block support to rate queue

### PR DESCRIPTION
**Pull request by kvokka**
Thank you for great gem!

* Add blocks support for `RateQueue` class, which allow to use this gem with custom metrics.
* Fix Ruby 2.7 warnings

Also:
Close #16

### CODE REVIEW

**Comment by kvokka** [comment]
@sbfaulkner can you please take a look? Thank you

**Comment by kvokka** [comment]
@sbfaulkner Thank you for the update. I rebased the changes on the current master.

Also, FYI, rubocop can not execute, cos https://shopify.github.io/ruby-style-guide/rubocop.yml is not avaliable.

File: lib/limiter/mixin.rb [hunk 1]
```diff
@@ -2,20 +2,13 @@
 
 module Limiter
   module Mixin
-    def limit_method(method, rate:, interval: 60)
-      queue = RateQueue.new(rate, interval: interval)
+    def limit_method(method, rate:, interval: 60, &b)
+      queue = RateQueue.new(rate, interval: interval, &b)
 
       mixin = Module.new do
-        if RUBY_VERSION < "2.7"
```
[the remark this answers is not in the document]
Comment (sbfaulkner) [reply lib/limiter/mixin.rb:9]: can we? (ie. since 2.6 is still < 2.7)

[the remark this answers is not in the document]
Comment (sbfaulkner) [reply lib/limiter/mixin.rb:9]: I *think* my testing showed that wouldn't work on 2.6

[the remark this answers is not in the document]
Comment (kvokka) [reply lib/limiter/mixin.rb:9]: Ruby 2.7 has deprecated automatic conversion from a hash to keyword arguments. 
We do not need to worry about 2.6 here, cos the syntax is valid for 2.0+

[Details](https://blog.saeloun.com/2019/10/07/ruby-2-7-keyword-arguments-redesign.html)

[the remark this answers is not in the document]
Comment (sbfaulkner) [reply lib/limiter/mixin.rb:9]: yeah... I get that... I just thought I'd tested on 2.6 and needed to handle differently... I may be misremembering :-)

[the remark this answers is not in the document]
Comment (kvokka) [reply lib/limiter/mixin.rb:9]: Np, just for sure i installed Ruby 2.6.3 locally and checked - everything is ok. 

But in the process of this check was revealed that travisci just does not work on the repo. Just letting you know

**Comment by kvokka** [comment]
Sorry @sbfaulkner , forgot to re-request the review.

**Comment by sbfaulkner** [comment]
@kvokka https://rubygems.org/gems/ruby-limiter/versions/2.1.0
🥂

**Pull request merged**